ibm_driver.c

```diff
@@ -736,25 +792,47 @@ static int ibm_handle_fetch_error(
 	add_next_index_long(info, conn_res->error_data.sqlcode);
         add_next_index_string(info, suppliment);
 
+#if !(PHP_MAJOR_VERSION > 8 || (PHP_MAJOR_VERSION == 8 && PHP_MINOR_VERSION == 1))
 	return TRUE;
+#endif
 }
 
 /* quotes an SQL statement */
+#if PHP_MAJOR_VERSION > 8 || (PHP_MAJOR_VERSION == 8 && PHP_MINOR_VERSION == 1)
+static zend_string* ibm_handle_quoter(
+	pdo_dbh_t *dbh,
+	const zend_string *unquoted,
+	enum pdo_param_type paramtype)
+#else
 static int ibm_handle_quoter(
 	pdo_dbh_t *dbh,
 	const char *unq,
 	size_t unq_len,
 	char **q,
 	size_t *q_len,
 	enum pdo_param_type paramtype)
+#endif
 {
 	char *sql;
 	size_t new_length, i, j;
+#if PHP_MAJOR_VERSION > 8 || (PHP_MAJOR_VERSION == 8 && PHP_MINOR_VERSION == 1)
+	/* keep the logic close to the pre-8.1 version w/ compat vars */
```

Comment:
I realize that's easier in the short term, but the better way is port to the new version and make the old code work w/ compat layers that way you can remove the compat layers when you drop the old versions. Otherwise, you end up like ibm_db2 where it was still trying to make things look like PHP 5.4 on 7.3.